servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -125,22 +130,25 @@
 
     private DefaultNettyConnection(Channel channel, BufferAllocator allocator, Executor executor,
                                    TerminalPredicate<Read> terminalMsgPredicate, CloseHandler closeHandler,
-                                   FlushStrategy flushStrategy, ExecutionStrategy executionStrategy) {
-        this(channel, allocator, executor, terminalMsgPredicate, closeHandler, flushStrategy, executionStrategy, null);
+                                   FlushStrategy flushStrategy, @Nullable Long idleTimeoutMs,
+                                   ExecutionStrategy executionStrategy) {
+        this(channel, allocator, executor, terminalMsgPredicate, closeHandler, flushStrategy, idleTimeoutMs,
+                executionStrategy, null);
     }
 
     private DefaultNettyConnection(Channel channel, BufferAllocator allocator, Executor executor,
                                    TerminalPredicate<Read> terminalMsgPredicate, CloseHandler closeHandler,
-                                   FlushStrategy flushStrategy, ExecutionStrategy executionStrategy,
-                                   @Nullable SSLSession sslSession) {
+                                   FlushStrategy flushStrategy, @Nullable Long idleTimeoutMs,
```

Comment:
I am somewhat apprehensive about individually adding options here which are otherwise not stored by netty. Anything else we add will mean we add more arguments here.

Can we instead take+store the `TcpConfig` here and pass the same to `SocketOptionUtils` for querying? If we do use `TcpConfig` we can use that to also get the `FlushStrategy`

Reply:
Thought about that but decided to defer until we add anything else to better understand how to structure it (maybe it will be better to have an object to keep these options instead of using the full `TcpConfig`).

We do not expose `AbstractReadOnlyTcpConfig` as a public class for now. If you think we can make `AbstractReadOnlyTcpConfig` public (it lives in `tcp-netty-internal` module) I can do the change now. Otherwise, let's defer?

We may also consider using a single `ExecutionContext` instead of other 3 separate arguments: `BufferAllocator`, `Executor`, `ExecutionStrategy`. Can do here or as a follow-up.

Reply:
Can we just use `TcpConfig` for now as these are internal classes and we can make sure we do not update config? If you want, we can also wrap the config to throw from the setters.

I am also OK to defer, your call.

Reply:
Will defer that, thanks